src/store/selectors/deck-filters.ts

```diff
@@ -0,0 +1,277 @@
+    default:
+      return (_: ResolvedDeck) => true;
+  }
+};
+
+// Exp Cost
+export const selectDecksMinMaxExpCost = createSelector(
+  selectLocalDecks,
+  (decks) => {
+    const minmax: RangeMinMax = decks.reduce<[number, number]>(
+      (acc, val) => {
+        const { xpRequired } = val.stats;
+        acc[0] = Math.min(acc[0], xpRequired);
+        acc[1] = Math.max(acc[1], xpRequired);
+        return acc;
+      },
+      [Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY],
+    );
+    return minmax;
+  },
+);
+
+export const selectExpCostChanges = createSelector(
+  selectDeckFilters,
+  (filters) => {
+    const expMinMax = filters.expCost;
+    return expMinMax ? `${expMinMax[0]}-${expMinMax[1]} experience` : "";
+  },
+);
+
+export const makeDeckExpCostFilter = (minmax: [number, number]) => {
+  return (deck: ResolvedDeck) => {
+    return (
+      deck.stats.xpRequired >= minmax[0] && deck.stats.xpRequired <= minmax[1]
+    );
+  };
+};
+
+const selectFilteringFunc = createSelector(selectDeckFilters, (filters) => {
+  const filterFuncs = [];
+  for (const filter of Object.keys(filters) as DeckFiltersKey[]) {
+    switch (filter) {
+      case "faction": {
+        const currentFilter = filters[filter];
+        if (currentFilter.length) {
+          filterFuncs.push(makeDeckFactionFilter(currentFilter));
+        }
+        break;
+      }
+      case "tags": {
+        const currentFilter = filters[filter];
+        if (currentFilter.length) {
+          filterFuncs.push(makeDeckTagsFilter(currentFilter));
+        }
+        break;
+      }
+      case "properties": {
+        const currentFilter = filters[filter];
+        filterFuncs.push(makeDeckPropertiesFilter(currentFilter));
+        break;
+      }
+      case "validity": {
+        const currentFilter = filters[filter];
+        if (currentFilter !== "all") {
+          filterFuncs.push(makeDeckValidityFilter(currentFilter));
+        }
+        break;
+      }
+      case "expCost": {
+        const currentFilter = filters[filter];
+        if (currentFilter) {
+          filterFuncs.push(makeDeckExpCostFilter(currentFilter));
+          break;
+        }
+      }
+    }
+  }
+
+  return and(filterFuncs);
+});
+
+export const selectDecksFiltered = createSelector(
+  selectLocalDecks,
+  selectDeckSearchTerm,
+  selectSearchableTextInDecks,
+  selectFilteringFunc,
+  (decks, searchTerm, searchableText, filterFunc) => {
+    let decksToFilter: ResolvedDeck[];
+
+    if (searchTerm) {
+      const finder = new uFuzzy({
+        intraMode: 0,
+        interIns: MATCHING_MAX_TOKEN_DISTANCE_DECKS,
+      });
+
+      const results = finder.search(searchableText, searchTerm);
+
+      decksToFilter = [];
+
+      if (results[0]) {
+        for (const result of results[0]) {
+          assert(decks[result], "Searching outside of bounds");
+          decksToFilter.push(decks[result]);
+        }
+      }
+    } else {
+      decksToFilter = [...decks];
```

Comment:
I think we don't need the spread here as we filter the decks in the next steps, which creates a new array